src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs

```diff
@@ -79,48 +81,52 @@ internal void WriteTo(ObjectWriter writer)
-                var builder = ImmutableArray.CreateBuilder<string>(inheritanceNamesLength);
-                for (var i = 0; i < inheritanceNamesLength; i++)
-                {
-                    builder.Add(reader.ReadString());
-                }
+            var name = reader.ReadString();
+            var immediateContainer = reader.ReadString();
+            var entireContainer = reader.ReadString();
+            var kind = (DeclaredSymbolInfoKind)reader.ReadByte();
+            var spanStart = reader.ReadInt32();
+            var spanLength = reader.ReadInt32();
+            var parameterCount = reader.ReadUInt16();
+            var typeParameterCount = reader.ReadUInt16();
 
-                return new DeclaredSymbolInfo(
-                    name, immediateContainer, entireContainer, kind, new TextSpan(spanStart, spanLength),
-                    builder.MoveToImmutable(),
-                    parameterCount, typeParameterCount);
-            }
-            catch
+            var inheritanceNamesLength = reader.ReadInt32();
+            var builder = ImmutableArray.CreateBuilder<string>(inheritanceNamesLength);
+            for (var i = 0; i < inheritanceNamesLength; i++)
             {
-                return default(DeclaredSymbolInfo);
+                builder.Add(reader.ReadString());
             }
+
+            return new DeclaredSymbolInfo(
+                name, immediateContainer, entireContainer, kind, new TextSpan(spanStart, spanLength),
+                builder.MoveToImmutable(), parameterCount, typeParameterCount);
         }
 
-        public async Task<ISymbol> ResolveAsync(Document document, CancellationToken cancellationToken)
+        public async Task<ISymbol> TryResolveAsync(Document document, CancellationToken cancellationToken)
         {
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            return Resolve(semanticModel, cancellationToken);
+            return TryResolve(semanticModel, cancellationToken);
         }
 
-        public ISymbol Resolve(SemanticModel semanticModel, CancellationToken cancellationToken)
+        public ISymbol TryResolve(SemanticModel semanticModel, CancellationToken cancellationToken)
         {
             var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
-            var node = root.FindNode(this.Span);
-            return semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            if (root.FullSpan.Contains(this.Span))
+            {
+                var node = root.FindNode(this.Span);
+                return semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            }
+            else
+            {
+                var message =
+$@"Invalid span in {nameof(DeclaredSymbolInfo)}.
+{nameof(this.Span)} = {this.Span}
+{nameof(root.FullSpan)} = {root.FullSpan}";
+                FatalError.ReportWithoutCrash(new InvalidOperationException(message));
```

Comment:
Do you think this gives you enough information to fix the issue from the reports?  Just knowing the two different spans?